internal/compiler/utils.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2023 The OPA Authors.  All rights reserved.
+// Use of this source code is governed by an Apache2
+// license that can be found in the LICENSE file.
+
+package compiler
+
+import (
+	"github.com/open-policy-agent/opa/ast"
+	"github.com/open-policy-agent/opa/schemas"
+	"github.com/open-policy-agent/opa/util"
+)
+
+type SchemaFile string
+
+const (
+	AuthorizationPolicySchema SchemaFile = "authorizationPolicy.json"
+)
+
+var schemaDefinitions = map[SchemaFile]interface{}{}
+
+func VerifyAuthorizationPolicySchema(rules []*ast.Rule) error {
+
+	if len(rules) == 0 {
+		return nil
+	}
+
+	parsed, err := ast.ParseRef("schema.input")
+	if err != nil {
+		return err
+	}
+
+	schemaSet := ast.NewSchemaSet()
+	schemaSet.Put(parsed, schemaDefinitions[AuthorizationPolicySchema])
+
+	parsedPath, err := ast.ParseRef("input")
+	if err != nil {
+		return err
+	}
+
+	sa := &ast.SchemaAnnotation{
+		Path:   parsedPath,
+		Schema: parsed,
+	}
+
+	c := ast.NewCompiler().WithSchemas(schemaSet).WithSchemaAnnotations([]*ast.SchemaAnnotation{sa})
+
+	errs := c.PassesTypeCheckRules(rules, nil)
+
+	if len(errs) > 0 {
+		return errs
+	}
+
+	return nil
+}
+
+func loadAuthorizationPolicySchema() {
+
+	ents, err := schemas.FS.ReadDir(".")
```

Comment:
Reading it again, I think my initial comment is a bit ambiguous. To clarify: I meant we could embed the file locations, not the file content itself. We'd still organize known schemas under a common folder, but since we still need to somehow map each file to where it should be applied. We already have an explicit reference to `authorizationPolicy.json`, and iterating through the entries in the FS seem redundant.
Not a biggie, though :).

Reply:
Ahh that makes sense. I'll look into that.

Reply:
Updated to read file directly and avoid reading the directory.